app/models/character.rb

```diff
@@ -31,4 +31,44 @@ def assign_skills
       skills.create(json_skill.attributes)
     end
   end
+
+  def upgrade_skills(skills_names)
+    self.skills.where(name: skills_names).update_all("rank = rank + 1")
+  end
+
+  def set_rank(skill, status)
+    if (self.career_skill?(skill.name))
+      if self.enough_xp?(skill.career_skill)
+        self.set_xp(skill, skill.career_skill, status)
+      end
+    else 
+      if self.enough_xp?(skill.normal_skill)
+        self.set_xp(skill, skill.normal_skill, status)
+      end
+    end
+  end
+
+  def career_skill?(skill_name)
+    self.career.career_skills.include?(skill_name)
+  end
+
+  def set_xp(skill, xp, operation)
+    if operation == "increment"
+      self.decrement!(:available_xp, by = xp)
+      self.increment!(:total_xp, by = xp)
+      skill.increment!(:rank, by = 1)
+    else
+      self.increment!(:available_xp, by = xp - 5)
+      self.decrement!(:total_xp, by = xp - 5)
+      skill.decrement!(:rank, by = 1)
+    end
+  end
+
+  def enough_xp?(xp_to_decrement)
+    if (self.available_xp - xp_to_decrement) > 0 
```

Comment:
![shits-on-fire-yo_o_153652](https://user-images.githubusercontent.com/236312/27747400-957300de-5d90-11e7-93cc-08db57009de0.jpg)

Why don't you just return the evaluation:

```ruby
def enough_xp?(xp_to_decrement)
  (self.available_xp - xp_to_decrement) > 0
end
```